x/gamm/keeper/pool_test.go

```diff
@@ -518,6 +518,83 @@ func (s *KeeperTestSuite) TestSetStableSwapScalingFactors() {
 	}
 }
 
+func (s *KeeperTestSuite) TestSetStableSwapScalingFactorController() {
+	initialControllerAddr := s.TestAccs[0].String()
+	updatedControllerAddr := s.TestAccs[1].String()
+
+	testcases := []struct {
+		name             string
+		poolId           uint64
+		expError         error
+		isStableSwapPool bool
+	}{
+		{
```

Comment:
Lets add a test case where we try to use an invalid address as the controller addr

Reply:
the address validation happens upstream in ValidateBasic. Want me to duplicate the check here as well?